runtime/tr.source/trj9/z/env/J9CPU.cpp

```diff
@@ -205,33 +206,17 @@ CPU::TO_PORTLIB_get390zLinuxMachineType()
    const int PROC_HEADER_SIZE = sizeof(procHeader) - 1;
 
 #if defined(J9ZTPF)
-   TR_S390MachineType ret_machine = TR_Z10;  /* return value, z/TPF default */
    struct pi1dt *pid;
-   /* machine hardware name */
+   /*
+    * Use the model number from the in-core processor ID table, accessed
+    * via CINFC_CMMPID as the machine type.
+    * The z/TPF OS has strict requirements on supported z/Architecture hardware.
+    * As a result, the model number will necessarily match a valid TR_S390MachineType.
+    * Note - S390MachineTypes, declared above as a static const integer, contains the
+    * hex representation of the (decimal) model number found in the processors ID table.
```

Comment:
Minor comment nit.  `S390MachineTypes` you described in an static const integer array containing the recognized machine types.   Are you referencing the `TR_S390MachineType` enum instead (https://github.com/eclipse/omr/blob/master/compiler/env/ProcessorInfo.hpp#L28-L61).  Also, not sure about the hex representation part... they are just binary integers. :)

Reply:
The value that z/TPF picks up - the model number - matches TR_S390MachineType but not the binary integer. For example, model 2827 (ec12) was loaded into a register and compared against the value in the memory location for S390MachineTypes which is xB0B.  I can take the comment out but I wanted anyone on the Pok. team to be aware of why the checks are removed from this version of J9CPU.cpp

Reply:
Hmm.. okay.   2827 == 0xB0B.... the representation in registers or memory should be identical in bits.  If you had left the checks in there, did you observe check failures?

Reply:
@joransiu I updated the code to do what I should have done from the start... sprintf the data from core as character data and then convert it via atoi(). z/os kinda had to do the same thing although the data is already sprintf()'ed into uname. Then the checks work; otherwise they would fail.